app/routes/application.ts

```diff
@@ -1,5 +1,21 @@
 import Route from '@ember/routing/route';
 
+import { service } from '@ember/service';
+
+import type DocumentService from 'frontend-validation-tool/services/document';
+import ENV from '../config/environment';
+
 export default class ApplicationRoute extends Route {
-  model() {}
+  @service declare document: DocumentService;
+
+  model() {
+    if (!ENV.CORS_PROXY_URL) {
+      const origin = window.location.origin;
+      const dispatchPath = '/communica';
+      this.document.setCorsProxy(`${origin}${dispatchPath}?url=`);
+      return;
+    }
+
+    this.document.setCorsProxy(ENV.CORS_PROXY_URL);
```

Comment:
shouldn't it be in an else block? you seem to override what u do at line 13

Reply:
Ah no Im returning on line 16 :)